dl/src/main/scala/com/intel/analytics/bigdl/optim/DistriOptimizer.scala

```diff
@@ -571,6 +614,25 @@ class DistriOptimizer[T: ClassTag] (
 
     model
   }
+
+  private def getLatestFile(fileName: String): String = {
+    val fl = new java.io.File(checkpointPath.get)
+    val files = fl.listFiles(new FilenameFilter {
+      override def accept(dir: File, name: String): Boolean = {
+        name.startsWith(fileName)
+      }
+    })
+
+    var lastMod = Long.MinValue
```

Comment:
@yiheng Any thoughts about create a sub directory with UUID in name to avoid model files with same name? If there is no concern, I will start work on it.